lib/usd/ui/layerEditor/mayaCommandHook.cpp

```diff
@@ -208,8 +208,13 @@ void MayaCommandHook::lockLayer(
     MayaUsd::LayerLockType lockState,
     bool                   includeSubLayers)
 {
+    // Per design, we refuse to change the lock state of system-locked
+    // layers through the UI.
+    if (MayaUsd::isLayerSystemLocked(usdLayer))
```

Comment:
This also needs to be conditioned whether we skip system locked files or not.

Reply:
Again, I don't understand what you mean. We never want to affect system-locked layer from thr UI.

Reply:
What I would suggest though, is to add the `skipSystemLocked`. Because unlike Muting, sub-layers of a system-locked layer can be unlocked.

Reply:
Ah, so we should et it do its work if include sub-layers is true? OK.

Reply:
Actually the code behind the lock layer and sublayer menu is
https://github.com/Autodesk/maya-usd/blob/7466596d572c800c68e37c587e8687ab11b534a0/plugin/adsk/scripts/mayaUsdMenu.mel#L945-L953

which seems to be enabled only if the right-clicked layer isn't system locked.